src/NLog/Layouts/Layout.cs

```diff
@@ -114,6 +114,28 @@ public static Layout FromString(string layoutText, ConfigurationItemFactory conf
             return new SimpleLayout(layoutText, configurationItemFactory);
         }
 
+        /// <summary>
+        /// Implicits converts the specific lambda method into a <see cref="SimpleLayout"/>.
+        /// </summary>
+        /// <param name="layoutMethod">Method that renders the layout.</param>
+        /// <param name="threadSafe">Tell if method is safe for concurrent threading.</param>
+        /// <returns>Instance of <see cref="SimpleLayout"/>.</returns>
+        public static Layout FromLayoutMethod(Func<LogEventInfo, object> layoutMethod, bool threadSafe = false)
```

Comment:
Consider replacing "threadSafe " with an object (e.g. LayoutCreationOptions). That would be easier to extend in the future)

Reply:
Have introduced parameter LayoutRenderOptions instead of `bool threadSafe`